typescript/phoenix-sdk/src/utils/seatManager.ts

```diff
@@ -181,23 +178,49 @@ export async function confirmOrCreateClaimSeatIxs(
 
   const instructions: TransactionInstruction[] = [];
 
-  const seatAccountInfo = await connection.getAccountInfo(seat, "confirmed");
+  let seatAccountInfo;
+  try {
+    seatAccountInfo = await connection.getAccountInfo(seat, "confirmed");
+  } catch {
+    seatAccountInfo = null;
+  }
+
   if (seatAccountInfo === null || seatAccountInfo.data.length == 0) {
     const traderToEvict = await findTraderToEvict(connection, market);
     if (traderToEvict) {
-      instructions.push(
-        getEvictSeatIx(market.address, market.data, traderToEvict, trader)
-      );
+      instructions.push(getEvictSeatIx(market, traderToEvict, trader));
     }
     instructions.push(getClaimSeatIx(market.address, trader));
   }
 
   return instructions;
 }
 
+/**
+ * Checks if the given trader has a seat on the given market
+ * If not, return claim seat instructions
+ * @param connection An instance of the Connection class
+ * @param market The market object
+ * @param trader The trader's address
+ */
+export async function createClaimSeatIdempotentInstructions(
```

Comment:
i thought we wanted the idempotent instruction functions to not be async? if they are async, why don't we just check everything up front?

Reply:
I don't think we can make this one sync? I can try playing around with it a bit

Reply:
We can make it sync, but it's a decent amount of work to pass in the seat manager struct to get the DMM list. I think we should punt

Reply:
I don't think we can do it either but then I think we shouldn't call this idempotent

Reply:
fixed